kinto/plugins/hawk/authentication.py

```diff
@@ -0,0 +1,50 @@
+from pyramid.interfaces import IAuthenticationPolicy
+from pyramid.authentication import CallbackAuthenticationPolicy
+from pyramid.authentication import extract_http_basic_credentials
+
+from zope.interface import implementer
+
+from kinto.core.storage import exceptions as storage_exceptions
+
+from .hawkauth import HawkAuth
+
+@implementer(IAuthenticationPolicy)
+class HawkAuthenticationPolicy(CallbackAuthenticationPolicy):
+    def unauthenticated_userid(self, request):
+        user_id = self._test_credentials(request)
+        return user_id
+
+    def forget(self, request):
+        return [('WWW-Authenticate', 'Hawk')]
+        
+    def _test_credentials(self, request):
+        """Test credentials in request against account HAWK credentials"""
+        account_creds = self._get_account_hawk_creds(request)
+        try:
+            return account_creds['id']
+        except:
+            return None
```

Comment:
why not merge `_test_credentials` and `_get_account_hawk_creds` methods?

Reply:
I must have had the idea that it was more modular, but looking at it again I think you are right-- it makes more sense as one method